src/OmniSharp.DotNetTest/VSTestManager.cs

```diff
@@ -184,10 +191,19 @@ public override GetTestStartInfoResponse GetTestStartInfo(string methodName, str
         }
 
         public override RunTestResponse RunTest(string methodName, string testFrameworkName, string targetFrameworkVersion)
+            => RunTest(new string[] { methodName }, testFrameworkName, targetFrameworkVersion);
+
+        public override RunTestResponse RunTest(string[] methodNames, string testFrameworkName, string targetFrameworkVersion)
         {
             VerifyTestFramework(testFrameworkName);
 
-            var testCases = DiscoverTests(methodName, targetFrameworkVersion);
+            var testCasesList = new List<TestCase>();
+            foreach (var methodName in methodNames)
+            {
+                testCasesList.AddRange(DiscoverTests(methodName, targetFrameworkVersion));
```

Comment:
This will cause test discovery to be performed for every method in the class, which is unnecessarily slow. Could you update `DiscoverTests` to take multiple method names?